dapps/profile.vue/src/components/did/ServiceEndpoints.ts

```diff
@@ -59,36 +68,78 @@ export default class ServiceEndpointsComponent extends mixins(EvanComponent) {
     },
   ]
 
-  addEndpointRow(): void {
-    const newEndpoint: ServiceEndpoint = {
-      id: this.newId,
-      type: this.newType,
-      url: this.newUrl,
-    };
+  get endpointIds(): string[] {
+    return this.endpoints.map((endpoint) => endpoint.id);
+  }
+
+  created(): void {
+    this.initValidation();
+  }
+
+  onSubmitRow(): void {
+    const newEndpoints: ServiceEndpoint[] = [
+      ...this.endpoints,
+      {
+        id: this.newId,
+        type: this.newType,
+        url: this.newUrl,
+      },
+    ];
+
     this.newId = null;
-    this.newUrl = null;
     this.newType = null;
+    this.newUrl = null;
+    // Workaround to reset to non-dirty inputs
+    this.formKey = new Date().toISOString();
 
-    this.$emit('addEndpoint', newEndpoint);
+    this.$emit('updateEndpoints', newEndpoints);
   }
 
   /**
    * Removes the selected endpoint temporarily
    * @param index row index of the item to be removed
    */
   deleteEndpoint(index: number): void {
-    this.$emit('deleteEndpoint', index);
-  }
-
-  editId(value: string, tableData: EvanTableItem<ServiceEndpoint>): void {
-    this.$emit('updateEndpoint', tableData.index, { id: value, type: tableData.item.type, url: tableData.item.url });
-  }
+    const newEndpoints = this.endpoints.filter((_, idx) => idx !== index);
 
-  editType(value: string, tableData: EvanTableItem<ServiceEndpoint>): void {
-    this.$emit('updateEndpoint', tableData.index, { id: tableData.item.id, type: value, url: tableData.item.url });
+    this.$emit('updateEndpoints', newEndpoints);
   }
 
-  editUrl(value: string, tableData: EvanTableItem<ServiceEndpoint>): void {
-    this.$emit('updateEndpoint', tableData.index, { id: tableData.item.id, type: tableData.item.type, url: value });
+  /**
+   * Set up used validation rules
+   */
+  initValidation(): void {
+    extend('required', {
+      ...required,
+      message: this.$t('_evan.validation.required'),
+    });
+
+    extend('url', {
+      validate: (value: string) => {
+        try {
+          /**
+           * Validate against RFC 3986
+           * Not pretty, but a regex would be even worse
+           */
+          // eslint-disable-next-line no-new
+          new URL(value);
+          return true;
```

Comment:
I think the URL constructor solution is nice, no excuses necessary ;)
Returning the boolean result should make the linter comment obsolete. 
Should work like this, but please test before merge.

```suggestion
        try {
          // Validate against RFC 3986
          return !!new URL(value);
```

Reply:
Why actually not just use the `url` validator from the veevalidate plugin?

Reply:
> Why actually not just use the url validator from the veevalidate plugin?
![image](https://user-images.githubusercontent.com/3518618/77887971-cf1be380-726b-11ea-913b-d2bb372781f7.png)
They don't seem to have one.